doc/source/reference/arrays.scalars.rst

```diff
@@ -96,84 +97,64 @@ Python Boolean scalar.
-
-Integers:
-
-===================  =============================  ===============
-:class:`byte`        compatible: C char             ``'b'``
-:class:`short`       compatible: C short            ``'h'``
-:class:`intc`        compatible: C int              ``'i'``
-:class:`int_`        compatible: Python int         ``'l'``
-:class:`longlong`    compatible: C long long        ``'q'``
-:class:`intp`        large enough to fit a pointer  ``'p'``
-:class:`int8`        8 bits
-:class:`int16`       16 bits
-:class:`int32`       32 bits
-:class:`int64`       64 bits
-===================  =============================  ===============
-
-Unsigned integers:
-
-===================  =============================  ===============
-:class:`ubyte`       compatible: C unsigned char    ``'B'``
-:class:`ushort`      compatible: C unsigned short   ``'H'``
-:class:`uintc`       compatible: C unsigned int     ``'I'``
-:class:`uint`        compatible: Python int         ``'L'``
-:class:`ulonglong`   compatible: C long long        ``'Q'``
-:class:`uintp`       large enough to fit a pointer  ``'P'``
-:class:`uint8`       8 bits
-:class:`uint16`      16 bits
-:class:`uint32`      32 bits
-:class:`uint64`      64 bits
-===================  =============================  ===============
-
-Floating-point numbers:
-
-===================  =============================  ===============
-:class:`half`                                       ``'e'``
-:class:`single`      compatible: C float            ``'f'``
-:class:`double`      compatible: C double
-:class:`float_`      compatible: Python float       ``'d'``
-:class:`longfloat`   compatible: C long float       ``'g'``
-:class:`float16`     16 bits
-:class:`float32`     32 bits
-:class:`float64`     64 bits
-:class:`float96`     96 bits, platform?
-:class:`float128`    128 bits, platform?
-===================  =============================  ===============
-
-Complex floating-point numbers:
-
-===================  =============================  ===============
-:class:`csingle`                                    ``'F'``
-:class:`complex_`    compatible: Python complex     ``'D'``
-:class:`clongfloat`                                 ``'G'``
-:class:`complex64`   two 32-bit floats
-:class:`complex128`  two 64-bit floats
-:class:`complex192`  two 96-bit floats,
-                     platform?
-:class:`complex256`  two 128-bit floats,
-                     platform?
-===================  =============================  ===============
-
-Any Python object:
-
-===================  =============================  ===============
-:class:`object_`     any Python object              ``'O'``
-===================  =============================  ===============
+.. inheritance-diagram:: byte short intc int_ longlong ubyte ushort uintc uint ulonglong half single double longdouble csingle cdouble clongdouble bool_ datetime64 timedelta64 object_ bytes_ str_ void
+
+Signed integer types
```

Comment:
Rather than `..autoclass:: numpy.byte`, try
```
.. autosummary::
    :toctree: generated
    :nosignatures:

    numpy.byte
```

Reply:
That would probably work, but I don't really want a page per class, because then we loose the ability to see the character codes and aliases all in one place.

Reply:
Then I think you need to create a template to override the default one in `doc/source/_templates/autosummary/class.rst` and use a `:template:` option on the [autosummary](https://www.sphinx-doc.org/en/master/usage/extensions/autosummary.html?highlight=templates#customizing-templates). I don't know if `:template:` works with `autoclass`

Reply:
Autosummary shouldn't be relevant here - it's built on top of autodoc, not vice versa. I think numpydoc is injecting an autosummary into the autodoc docstring or something.